src/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,177 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { IconDefinition } from "@fortawesome/free-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { ThemeContext, ThemeProvider } from "../../theme";
+import { isNavbarItem } from "./Navbar.utils";
+import { isDev } from "../../utils";
+import useDotAnimation from "./hooks";
+import style from "../../variables.module.scss";
+import "./navbar.scss";
+
+const NavbarContext = React.createContext<{
+  items?: React.RefObject<HTMLLIElement>[];
+  activeId?: number;
+  setActiveId?: (id?: number) => void;
+  registerItem?: (item: React.RefObject<HTMLLIElement>) => void;
+  unRegisterItem?: (item: React.RefObject<HTMLLIElement>) => void;
+}>({});
+
+type NavbarItemProps = {
+  icon: IconDefinition;
+  title: string;
+  onSelect: ({ title, id }: { title: string; id?: number }) => void;
+  id?: number; // TODO: find out how not to expose this prop
```

Comment:
question: Why do you need to expose this prop?

Reply:
I'm making TS happy 😅

Reply:
Ah nevermind, I misunderstood this line 🙈